libsyclinterface/source/dpctl_sycl_device_interface.cpp

```diff
@@ -849,3 +849,57 @@ DPCTLDevice_GetSubGroupSizes(__dpctl_keep const DPCTLSyclDeviceRef DRef,
     }
     return sizes;
 }
+
+__dpctl_give DPCTLDeviceVectorRef
+DPCTLDevice_GetComponentDevices(__dpctl_keep const DPCTLSyclDeviceRef DRef)
+{
+    using vecTy = std::vector<DPCTLSyclDeviceRef>;
+    vecTy *ComponentDevicesVectorPtr = nullptr;
```

Comment:
You can use `unique_ptr` here to avoid calling `delete` by hands. 

```C++
auto ComponentDevicesVectorPtr = std::unique_ptr<vecTy>();
if (DRef) {
    ...
    try {
        ...
        ComponentDevicesVectorPtr.reset(new VecTy());
        ...
    }
    catch(...) {
            error_handler(e, __FILE__, __func__, __LINE__);
            return nullptr;
    }


return wrap<vecTy>(ComponentDevicesVectorPtr.release());

Reply:
This is something that might be smart to go back and do throughout the libsyclinterface codebase